[SQL] Generate SDK for Endpoint Certificates API

ostojicb opened the pull request:
# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included.
https://github.com/Azure/azure-rest-api-specs/pull/18502
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

ostojicb committed:
initial generate

ostojicb committed:
Add tests

ostojicb committed:
Merge branch 'main' into sqlmi_endpoint_cert

ostojicb reviewed (commented):
swagger PR merged https://github.com/Azure/azure-rest-api-specs/pull/18502

ostojicb committed:
run generate.ps1 script

ostojicb committed:
Merge branch 'main' into sqlmi_endpoint_cert

ostojicb committed:
Bump versions

ostojicb commented:
Hi @m-nash, can you take a look?

archerzz commented on sdk/sqlmanagement/Microsoft.Azure.Management.Sql/src/Generated/IReplicationLinksOperations.cs:
```diff
@@ -272,13 +235,15 @@ public partial interface IReplicationLinksOperations
         /// <exception cref="Microsoft.Rest.ValidationException">
         /// Thrown when a required parameter is null
         /// </exception>
-        Task<AzureOperationResponse> BeginFailoverWithHttpMessagesAsync(string resourceGroupName, string serverName, string databaseName, string linkId, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
```
breaking change

archerzz commented on sdk/sqlmanagement/Microsoft.Azure.Management.Sql/src/Generated/IReplicationLinksOperations.cs:
```diff
@@ -216,7 +181,7 @@ public partial interface IReplicationLinksOperations
         /// Thrown when a required parameter is null
         /// </exception>
-        Task<AzureOperationResponse<IPage<ReplicationLink>>> ListByDatabaseWithHttpMessagesAsync(string resourceGroupName, string serverName, string databaseName, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
+        Task<AzureOperationResponse<ReplicationLink>> FailoverAllowDataLossWithHttpMessagesAsync(string resourceGroupName, string serverName, string databaseName, string linkId, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
```
breaking change

archerzz commented on sdk/sqlmanagement/Microsoft.Azure.Management.Sql/src/Generated/IReplicationLinksOperations.cs:
```diff
@@ -149,46 +141,16 @@ public partial interface IReplicationLinksOperations
         /// <exception cref="Microsoft.Rest.ValidationException">
         /// Thrown when a required parameter is null
         /// </exception>
-        Task<AzureOperationResponse> UnlinkWithHttpMessagesAsync(string resourceGroupName, string serverName, string databaseName, string linkId, UnlinkParameters parameters, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
```
Breaking change: `UnlinkWithHttpMewwageAsync` is removed

archerzz commented on sdk/sqlmanagement/Microsoft.Azure.Management.Sql/src/Generated/Models/DatabaseUpdate.cs:
```diff
@@ -182,11 +182,8 @@ public DatabaseUpdate()
-        /// internal use only</param>
         /// <param name="tags">Resource tags.</param>
-        public DatabaseUpdate(Sku sku = default(Sku), DatabaseIdentity identity = default(DatabaseIdentity), string createMode = default(string), string collation = default(string), long? maxSizeBytes = default(long?), string sampleName = default(string), string elasticPoolId = default(string), string sourceDatabaseId = default(string), string status = default(string), System.Guid? databaseId = d ...
+        public DatabaseUpdate(Sku sku = default(Sku), DatabaseIdentity identity = default(DatabaseIdentity), string createMode = default(string), string collation = default(string), long? maxSizeBytes = default(long?), string sampleName = default(string), string elasticPoolId = default(string), string sourceDatabaseId = default(string), string status = default(string), System.Guid? databaseId = d ...
```
breaking change, a property is removed

archerzz reviewed (changes requested):
A few breaking changes are found. It seems that some of them do not have backward compatibility (like removal of property). Please bump the major version to `3.0.0.0`.

ostojicb committed:
revert collateral damage

ostojicb committed:
Merge branch 'main' into sqlmi_endpoint_cert

ostojicb committed:
Merge branch 'main' into sqlmi_endpoint_cert

ostojicb committed:
Merge branch 'main' into sqlmi_endpoint_cert

ostojicb committed:
run generate.ps1, take 2

ostojicb commented on sdk/sqlmanagement/Microsoft.Azure.Management.Sql/src/Generated/Models/IPv6FirewallRuleList.cs:
```diff
@@ -1,51 +0,0 @@
-// <auto-generated>
```
Introduced with https://github.com/Azure/azure-rest-api-specs/pull/18490 @LeiWang3 can you confirm this is not a breaking change (and other IPv6 related changes in this pull request)

ostojicb reviewed (commented):
Regarding the removal of `delegatedResources` and `primaryDelegatedIdentityClientId` in `Database*` files, this change has been applied to the API version that's **currently** in use - which means this change was already implicitly propagated to SDK and downstream clients. 

For that reason, I am not sure if this SDK update itself is a breaking one (whether it requires a major version bump) Also, I couldn't find any references to these classes/props in PowerShell code.

For the reference, this is the PR that changed the current API version https://github.com/Azure/azure-rest-api-specs/pull/18209
cc: @viparek

ostojicb committed:
Merge branch 'main' into sqlmi_endpoint_cert

ostojicb committed:
revive replication links

ostojicb committed:
Merge branch 'main' into sqlmi_endpoint_cert

ostojicb committed:
run generate.ps1

ostojicb committed:
bump to 3.0.0

ostojicb commented:
> A few breaking changes are found. It seems that some of them do not have backward compatibility (like removal of property). Please bump the major version to `3.0.0.0`.

We've reverted some breaking changes (from swagger). The one that's left (`Delegation*` stuff) is referring to already removed internal fields in the **current** API version - not sure if requires major version bump (3.0.0.0), but I updated it regardless.

archerzz commented:
Thanks @ostojicb  Can you bump `AssemblyVersion` to `3.0.0.0` as well? See https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/78/Mgmt-.NET-SDK-release-process?anchor=signing-and-publishing-nuget-packges

ostojicb committed:
Merge branch 'main' into sqlmi_endpoint_cert

ostojicb committed:
Bump AssemblyVersion
